### PR TITLE
pee

### PR DESCRIPTION
**Pull request by ishan-renjen**
finished code to read temperatures from tmp117

### CODE REVIEW

File: include/TMS/dev/TMP117.hpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+//
+// Created by Ishan on 1/23/2023.
```
Comment (mjmagee991) [inline include/TMS/dev/TMP117.hpp]: Delete this comment

File: include/TMS/dev/TMP117.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+//
+
+#include "EVT/io/I2C.hpp"
```
Comment (mjmagee991) [inline include/TMS/dev/TMP117.hpp]:
```suggestion
#include <EVT/io/I2C.hpp>
```

File: include/TMS/dev/TMP117.hpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+#define TMS_TMP117_HPP
+#endif//TMS_TMP117_HPP
+
+namespace IO = EVT::core::IO;
+
+namespace TMS {
+
+class TMP117 {
+public:
+    TMP117(IO::I2C& i2c, uint8_t deviceID);
```
Comment (mjmagee991) [inline include/TMS/dev/TMP117.hpp:24]: Add docstrings to these methods and the member variables

File: include/TMS/dev/TMP117.hpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+#ifndef TMS_TMP117_HPP
+#define TMS_TMP117_HPP
+#endif//TMS_TMP117_HPP
+
+namespace IO = EVT::core::IO;
+
+namespace TMS {
+
+class TMP117 {
```
Comment (mjmagee991) [inline include/TMS/dev/TMP117.hpp:16]: Add a docstring for the class

File: include/TMS/dev/TMP117.hpp [hunk 5]
```diff
@@ -0,0 +1,28 @@
+//
+// Created by Ishan on 1/23/2023.
+//
+
+#include "EVT/io/I2C.hpp"
+
+#ifndef TMS_TMP117_HPP
+#define TMS_TMP117_HPP
+#endif//TMS_TMP117_HPP
```
Comment (mjmagee991) [inline include/TMS/dev/TMP117.hpp]: This is what's called a header guard. The first two lines should be the first thing in the file, and the `#endif` should be the last thing in the file

File: src/TMS/dev/TMP117.cpp [hunk 6]
```diff
@@ -0,0 +1,29 @@
+//
+// Created by Ishan on 1/23/2023.
```
Comment (mjmagee991) [inline src/TMS/dev/TMP117.cpp]: Delete this comment

File: include/TMS/dev/TMP117.hpp [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    uint16_t readTemp();
+
+private:
+    constexpr static uint8_t* tempReg = 0x00;
+    uint8_t deviceID;
+    IO::I2C& i2c;
+    uint8_t regLength = 1;
+    uint8_t valueLength = 2;
+    uint16_t temp;
+};
```
Comment (mjmagee991) [inline include/TMS/dev/TMP117.hpp:46]:
```suggestion
};

```

File: include/TMS/dev/TMP117.hpp [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+public:
+    TMP117(IO::I2C& i2c, uint8_t deviceID);
+    uint16_t readTemp();
+
+private:
+    constexpr static uint8_t* tempReg = 0x00;
+    uint8_t deviceID;
+    IO::I2C& i2c;
+    uint8_t regLength = 1;
+    uint8_t valueLength = 2;
```
Comment (mjmagee991) [inline include/TMS/dev/TMP117.hpp:43]: `regLength` and `valueLength` can both be `constexpr static` variables because they're the same for all TMP117s

File: src/TMS/dev/TMP117.cpp [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+uint16_t TMP117::readTemp() {
+    uint16_t input;
+    uint8_t tempBytes[2];
+    uint16_t byte1;
+
+    i2c.readReg(deviceID, tempReg, regLength, tempBytes, valueLength);
+
+    byte1 = tempBytes[0];
+    byte1<<=8;
+    input = byte1|tempBytes[1];
```
Comment (mjmagee991) [inline src/TMS/dev/TMP117.cpp]:
```suggestion
    input = ((uint16_t) tempBytes[0]) << 8 | tempBytes[1];
```
If you do it this way, you can get rid of the variable `byte1`

File: src/TMS/dev/TMP117.cpp [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+    //degrees centi celsius???
+    temp = ((uint64_t)input)*78125/100000;
+
+    return temp;
+}
+
+}
```
Comment (mjmagee991) [inline src/TMS/dev/TMP117.cpp]: Add a newline to the end of the file

File: src/TMS/dev/TMP117.cpp [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    uint8_t tempBytes[2];
+    uint16_t byte1;
+
+    i2c.readReg(deviceID, tempReg, regLength, tempBytes, valueLength);
+
+    byte1 = tempBytes[0];
+    byte1<<=8;
+    input = byte1|tempBytes[1];
+
+    //degrees centi celsius???
```
Comment (mjmagee991) [inline src/TMS/dev/TMP117.cpp]: Add an explanation of how we got the numbers 78125 and 100000

File: targets/tmpSample/CMakeLists.txt [hunk 12]
```diff
@@ -0,0 +1,7 @@
+include(${EVT_CORE_DIR}/cmake/evt-core_build.cmake)
+
+project(tmpSample)
+cmake_minimum_required(VERSION 3.15)
+
+make_exe(${PROJECT_NAME} main.cpp)
+target_link_libraries(${PROJECT_NAME} PUBLIC ${BOARD_LIB_NAME})
```
Comment (mjmagee991) [inline targets/tmpSample/CMakeLists.txt:7]: Add a newline to the end of the file

File: targets/tmpSample/main.cpp [hunk 13]
```diff
@@ -0,0 +1,24 @@
+//
+// Created by Ishan on 1/30/2023.
+//
+#include "EVT/io/i2c.hpp"
```
Comment (mjmagee991) [inline targets/tmpSample/main.cpp]:
```suggestion
#include <EVT/io/i2c.hpp>
```
All the includes should use `<>`

File: targets/tmpSample/main.cpp [hunk 14]
```diff
@@ -0,0 +1,24 @@
+//
+// Created by Ishan on 1/30/2023.
```
Comment (mjmagee991) [inline targets/tmpSample/main.cpp]: Change this comment use `/**  */` and explain the use of this sample

File: targets/tmpSample/main.cpp [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+    TMS::TMP117 tmp = TMS::TMP117(i2c, 0x48);
+
+    while(1){
+        uart.printf("temp values: ");
+        uart.printf("%#d\n\r", tmp.readTemp());
+
+
+    }
+
+}
```
Comment (mjmagee991) [inline targets/tmpSample/main.cpp]: Add a newline to the end of the file

File: include/TMS/dev/TMP117.hpp [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace TMS {
+
+class TMP117 {
+public:
+    TMP117(IO::I2C& i2c, uint8_t deviceID);
+    uint16_t readTemp();
+
+private:
+    constexpr static uint8_t* tempReg = 0x00;
```
Comment (mjmagee991) [inline include/TMS/dev/TMP117.hpp]: This should just be a `uint8_t`, not a pointer

File: src/TMS/dev/TMP117.cpp [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+namespace TMS {
+
+TMP117::TMP117(IO::I2C& i2c, uint8_t deviceID): deviceID(deviceID), i2c(i2c) { }
+
+uint16_t TMP117::readTemp() {
+    uint16_t input;
+    uint8_t tempBytes[2];
+    uint16_t byte1;
+
+    i2c.readReg(deviceID, tempReg, regLength, tempBytes, valueLength);
```
Comment (mjmagee991) [inline src/TMS/dev/TMP117.cpp]:
```suggestion
    i2c.readReg(deviceID, &tempReg, regLength, tempBytes, valueLength);
```
When you change `tempReg` to a `uint8_t`, this will need get its address with `&`

File: targets/tmpSample/main.cpp [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+int main() {
+    EVT::core::platform::init();
+
+    IO::I2C& i2c = IO::getI2C<IO::Pin::PB_8, IO::Pin::PB_9>();
+    IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+    TMS::TMP117 tmp = TMS::TMP117(i2c, 0x48);
+
+    while(1){
+        uart.printf("temp values: ");
+        uart.printf("%#d\n\r", tmp.readTemp());
```
Comment (mjmagee991) [inline targets/tmpSample/main.cpp]:
```suggestion
        uart.printf("%d\n\r", tmp.readTemp());
```

File: targets/tmpSample/main.cpp [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+#include <EVT/io/pin.hpp>
+
+int main() {
+    EVT::core::platform::init();
+
+    IO::I2C& i2c = IO::getI2C<IO::Pin::PB_8, IO::Pin::PB_9>();
+    IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+    TMS::TMP117 tmp = TMS::TMP117(i2c, 0x48);
+
+    while(1){
```
Comment (mjmagee991) [inline targets/tmpSample/main.cpp]: It looks like the autoformatter didn't run on your PR. Be sure that it does run before you merge it into main

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Just some minor documentation and organization fixes

**Comment by mjmagee991** [comment]
[TMP117 Driver](https://trello.com/c/VVtADxfP/110-tmp117-driver)

**Review by ActuallyTaylor — APPROVED** [review]

**Review by mjmagee991 — APPROVED** [review]
Nice job

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
There are some old comments that got missed

**Review by mjmagee991 — APPROVED** [review]
Looks good

File: CMakeLists.txt [hunk 20]
```diff
@@ -54,7 +55,6 @@ if(NOT TARGET_DEV)
     set(TARGET_DEV "STM32F302x8")
 endif()
 
-
```
Comment (pro-sper) [inline CMakeLists.txt:57]: Unnecessary whitespace

**Review by pro-sper — APPROVED** [review]
